storybookjs/storybook
# UI: improve A11Y remove redundant styling rules, update icon color

Closes #23455






## What I did



Updated CSS for the accordion toggle icon within the accessibility pane.

- removed the muted, low contrast colour rule, colour is now the same as the text (#2E3438). This colour is cascaded to the icon from the body.
- removed redundant flex-box rules
- adjusted the alignment so the icon is centre aligned with the text.

CSS of the icon now looks as follows: 
```
.css-1eq8i2q {
  display: inline-block;
  shape-rendering: inherit;
  fill: currentColor;
  height: 10px;
  width: 10px;
  min-width: 10px;
  margin-right: 10px;
  transition: transform 0.1s ease-in-out;
  vertical-align: inherit;
}
```

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:
- [ x ] unit tests

#### Manual testing

To manually test this change, run a sandbox with the a11y add-on enabled. 
Using the colour contrast checker[https://webaim.org/resources/contrastchecker/] verify that the arrow icon has sufficient colour contrast.

![image](https://github.com/storybookjs/storybook/assets/10471661/08ba164c-0512-440c-a37e-b2195f9f3708)


1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-webpack/18-ts`
2. Within the sandbox `yarn add @storybook/addon-a11y --dev`
3. add  '@storybook/addon-a11y', to the addons array in `storybook/sandbox/react-webpack-18-ts/.storybook/main.ts`
4. `yarn storybook`
5. Open Storybook in your browser
6. Access any story
7. In the Addons panel check the accessibility addon panel
8. Verify the accordion toggle icon has sufficient colour contrast

![image](https://github.com/storybookjs/storybook/assets/10471661/eb720819-58ca-47a6-a2ff-47f2557f89f2)

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
